scripts/deploy_teleporter.sh

```diff
@@ -74,13 +85,21 @@ echo "TeleporterMessenger $teleporter_version contract address: $teleporter_cont
 teleporter_deployer_address=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Deployer_Address_$teleporter_version.txt)
 echo "TeleporterMessenger $teleporter_version deployer address: $teleporter_deployer_address"
 teleporter_deploy_tx=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Deployment_Transaction_$teleporter_version.txt)
+teleporter_messenger_bytecode=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Bytecode_$teleporter_version.txt)
 
 # Check if this TeleporterMessenger version has already been deployed on this chain.
-teleporter_contract_code=$(cast code --rpc-url $rpc_url $teleporter_contract_address)
-if [[ $teleporter_contract_code != "0x" ]]; then
+teleporter_contract_code=$(cast codesize $teleporter_contract_address --rpc-url $rpc_url)
+if [[ $teleporter_contract_code != "0" ]]; then
     echo "TeleporterMessenger $teleporter_version has already been deployed on this chain." && exit 0
 fi
 
+# Estimate the amount of gas required to deploy the TeleporterMessenger bytecode from the Teleporter
+# deployer address in order to simulate the transaction. This will error if the TeleporterMessenger
+# contract is unable to be deployed from the deployer address.
+cast estimate --rpc-url $rpc_url \
+    --from $teleporter_deployer_address \
+    --create $teleporter_messenger_bytecode > /dev/null
```

Comment:
Any particular reason to suppress the output here?

In the error case, we should make sure this emits a clear error. The `cast` error might already be clear enough, but we may want to explicitly check the output and `echo` an error message of our own.

Reply:
In the success case, I didn't think it was necessary to show the gas estimation output because the actual gas price used by Nick's method transaction is fixed since it's a legacy transaction, so I opted to suppress the stdout output.

The deployer allowlist error message was definitely clear enough from cast (seen below). I'm not 100% sure what other error cases may pop up, but another one that is helpful to have here is if the provided RPC URL is unreachable or invalid.

```
(code: -32000, message: tx.origin 0xa21D65b03A07548F001b6C1422614BEa28370FdA is not authorized to deploy a contract, data: None)
```

```
$ ./scripts/deploy_teleporter.sh --version v0.1.0 --rpc-url https://notablockchainendpoint.com
TeleporterMessenger v0.1.0 contract address: 0x1b0505EaC728Efc30414ce0035BcC4DC5994a563
TeleporterMessenger v0.1.0 deployer address: 0xa21D65b03A07548F001b6C1422614BEa28370FdA
Error: 
error sending request for url (https://notablockchainendpoint.com/): error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known

Context:
- Error #0: error trying to connect: dns error: failed to lookup address information: nodename nor servname provided, or not known
- Error #1: dns error: failed to lookup address information: nodename nor servname provided, or not known
- Error #2: failed to lookup address information: nodename nor servname provided, or not known
```

Reply:
Yeah I think this redirection should be fine. We're only redirecting `stdout`, not `stderr`, so we will still see any errors printed by `cast`

Reply:
Cool, agreed those error messages seem clear enough. I'm not seeing the line in the script in those messages though. That would be helpful to have.

Reply:
Discussed with Michael offline, there's no easy workaround to emit the failing line number. In most cases it should be obvious from the error message itself what failed.